pkg/operator/ceph/csi/spec.go

```diff
@@ -521,11 +526,12 @@ func deleteCSIDriverResources(
 	return succeeded
 }
 
-func applyCephClusterNetworkConfig(objectMeta *metav1.ObjectMeta, rookclientset rookclient.Interface) (bool, error) {
+func applyCephClusterNetworkConfig(ctx *clusterd.Context, objectMeta *metav1.ObjectMeta) (bool, error) {
 	var isMultusApplied bool
-	cephClusters, err := rookclientset.CephV1().CephClusters(objectMeta.Namespace).List(metav1.ListOptions{})
+	cephClusters := &v1.CephClusterList{}
+	err := ctx.Client.List(context.TODO(), cephClusters, &client.ListOptions{})
 	if err != nil {
-		return false, errors.Errorf("failed to find CephClusters in namespace %q", objectMeta.Namespace)
+		return false, errors.Errorf("failed to find CephClusters")
 	}
 	for _, cephCluster := range cephClusters.Items {
```

Comment:
will this list ceph cluster CRD belongs to some other rook operator (if two rook instances are running in a single kubernetes cluster?)

Reply:
Correct, it is possible to run multiple instances. The operator will either watch all namespaces or only the same namespace as the operator for clusters, and it is controlled by [this setting](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/operator.yaml#L297). However, I'm not sure any is actually running multiple operators, or at least it seems likely to be bugs if someone did try that...